sys-apps/util-linux/patches/util_linux-2.32.patchset

```diff
@@ -130,3 +125,43 @@ index e1c4059..ac553bf 100644
 -- 
 2.14.1
 
+
+From e8ab90b9e91df50304e32a77325d2030c1d4aeb6 Mon Sep 17 00:00:00 2001
+From: fbrosson <fbrosson@localhost>
+Date: Tue, 1 May 2018 23:37:15 +0000
+Subject: define fake makedev, major and minor functions.
+
+
+diff --git a/libblkid/src/blkidP.h b/libblkid/src/blkidP.h
+index 22c9856..719bd83 100644
+--- a/libblkid/src/blkidP.h
++++ b/libblkid/src/blkidP.h
+@@ -341,6 +341,10 @@ UL_DEBUG_DECLARE_MASK(libblkid);
+ #define UL_DEBUG_CURRENT_MASK	UL_DEBUG_MASK(libblkid)
+ #include "debugobj.h"
+ 
++extern unsigned int major(dev_t devno);
++extern unsigned int minor(dev_t devno);
++extern dev_t makedev(unsigned int ma, unsigned int mi);
++
+ extern void blkid_debug_dump_dev(blkid_dev dev);
+ 
+ 
+diff --git a/libblkid/src/dev.c b/libblkid/src/dev.c
+index 056df2b..3cbe939 100644
+--- a/libblkid/src/dev.c
++++ b/libblkid/src/dev.c
+@@ -15,6 +15,10 @@
+ 
+ #include "blkidP.h"
+ 
++unsigned int major(dev_t devno) { return devno>>8; }
++unsigned int minor(dev_t devno) { return devno & 0xff; }
++dev_t makedev(unsigned int ma, unsigned int mi) { return (dev_t)(ma<<8) | mi; }
```

Comment:
```diff
++unsigned int major(dev_t devno) { return devno>>8; }
++unsigned int minor(dev_t devno) { return devno & 0xff; }
++dev_t makedev(unsigned int ma, unsigned int mi) { return (dev_t)(ma<<8) | mi; }
```
Would it be acceptable to move these functions (or variants of it) to one of the system shared libraries or to libbsd, libgnu or anything more appropriate ?

Would someone be willing to take care of it?
@alaviss, since you are an expert, what would you recommend? Thanks!

Reply:
A quick look shows that those are common POSIX extensions that exists in the BSDs and Linux, so I would recommend placing them in `libroot`, implemented as macros. In fact, they are already in [Haiku's ntfs driver][0], so we just need to move them out.

[0]: http://xref.plausible.coop/source/xref/haiku/src/add-ons/kernel/file_systems/ntfs/libntfs/config.h#312

Reply:
Cool, thanks! I'll try with Haiku's ntfs driver.